tools/d2g/d2g.go

```diff
@@ -372,6 +371,10 @@ func setMounts(gwPayload *genericworker.GenericWorkerPayload, gwWritableDirector
 
 func setMaxRunTime(dwPayload *dockerworker.DockerWorkerPayload, gwPayload *genericworker.GenericWorkerPayload) {
 	gwPayload.MaxRunTime = dwPayload.MaxRunTime
+	if len(gwPayload.Artifacts) > 0 {
+		// Add 15 minutes as buffer for task to be able to upload artifacts
+		gwPayload.MaxRunTime += 900
```

Comment:
This mimics Queue behavior, I believe. A task that has been resolved can only upload artifacts for 15 minutes afterwards. In our case, we can be reasonably confident that the 15 mins won't actually be needed/used, because the timeout should have killed the long running container, and only the podman cp commands are running for the remainder of the time. So really it is just a safe upper bound, but doesn't mean that tasks will actually run that long - in practice, they should never run for the full 15 mins, and probably will just run for an extra 5-10 seconds or so.

Reply:
Perfect, thanks for this information!